### DISCUSSION TITLE
What version of PDDL and what requirements are supported?

### DISCUSSION THREAD

**Discussion by nvitucci** [aibasel/pyperplan#23] [Q&A] [answered]
What version of PDDL is supported? I'm thinking 1.2, but could you please confirm?
Related: what [requirements](https://planning.wiki/ref/pddl/requirements) are supported?

**Comment by maltehelmert** [comment, marked as answer]
Pyperplan supports the STRIPS fragment of PDDL.

This fragment hasn't really changed since PDDL 1.2; all later version of the language added features that need additional requirements. So it's as correct to say that pyperplan supports PDDL 1.2 as saying that it supports PDDL 3.0, but the important point is that it only supports the STRIPS fragment, which is a subset of level 1 of the language. When looking for a PDDL spec, it's probably a good idea to look at a more modern description than the technical report on PDDL 1.2. I recommend looking at descriptions of PDDL 2.1 or later; the differences between later versions like 2.1, 2.2 and 3.0 are very minor as long as you don't look at the advanced requirements (not supported by pyperplan) introduced in the later versions.

The requirements page you linked is not up to date; these are the requirements from 1998, many of which are abandoned (e.g. :ucpop or the ones related to HTN planning), while others were added. Of the ones that are considered to be part of current PDDL, pyperplan supports :strips and :typing. Perhaps also :negation and :equality; about this I am unsure.

**Reply by nvitucci** [reply]
Thanks for the clarification. I tried to use `:negative-preconditions` and it failed, hence my question.
Given that only `:strips` and `:typing` are supported, I take that numeric fluents and durative actions from 2.1+ are excluded as well.
